kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -75,6 +75,7 @@ public class FilterHandler extends ChannelDuplexHandler {
     private CompletableFuture<Void> readFuture = CompletableFuture.completedFuture(null);
     private @Nullable ChannelHandlerContext ctx;
     private @Nullable PromiseFactory promiseFactory;
+    private static boolean deprecationWarningEmitted = false;
```

Comment:
@tombentley did you mean a single log per process, or a single log per something else, per channel? I figured you meant globally since per-channel wouldn't really prevent much logging, just reauthentications.